### PR TITLE
Add note about empty fields and null values for API responses

### PR DESCRIPTION
**Pull request by umohnani8**
[NO TESTS NEEDED]

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

#### What this PR does / why we need it:

Add a note the global swagger docs about some fields not showing
up in responses as they are set to omitempty. Also add a note about
null values for complicated field types that swagger-go has a hard time
with.

#### How to verify it

Check swagger docs for update

#### Which issue(s) this PR fixes:

Fixes https://github.com/containers/podman/issues/11826
Fixes https://github.com/containers/podman/issues/11830

### CODE REVIEW

**Comment by umohnani8** [comment]
@jwhonce @TomSweeneyRedHat PTAL

File: pkg/api/server/docs.go [hunk 1]
```diff
@@ -15,6 +15,15 @@
 // NOTE: if you install the package podman-docker, it will create a symbolic
 // link for /run/docker.sock to /run/podman/podman.sock
 //
+// NOTE: some fields in the API response JSON are set as omitempty, which means that
+// if there is no value set for them, they will not show up in the response. This
+// is a feature to help reduce the size of the JSON responses returned.
+//
+// NOTE: due to the limitations of swagger-go, some field values that have a complex type
```
Comment (jwhonce) [inline pkg/api/server/docs.go:22]:
```suggestion
// NOTE: due to the limitations of [go-swagger](https://github.com/go-swagger/go-swagger), some field values that have a complex type
```

**Review by jwhonce — COMMENTED** [review]
One small nit

**Comment by jwhonce** [comment]
LGTM

File: pkg/api/server/docs.go (hunk 1 above)
Comment (umohnani8) [reply pkg/api/server/docs.go]: Fixed

**Comment by TomSweeneyRedHat** [comment]
one suspected flake test, I restarted.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**